skbio/stats/tests/test_composition.py

```diff
@@ -550,7 +550,131 @@ def test_ancom_basic_counts(self):
+        self.assertAlmostEqual(result[(0.0, 'a')]['b1'], 9.0)
+        self.assertAlmostEqual(result[(0.0, 'b')]['b1'], 20.0)
+        self.assertAlmostEqual(result[(0.0, 'c')]['b1'], 1.0)
+        self.assertAlmostEqual(result[(25.0, 'a')]['b1'], 9.75)
+        self.assertAlmostEqual(result[(25.0, 'b')]['b1'], 20.5)
+        self.assertAlmostEqual(result[(25.0, 'c')]['b1'], 6.5)
+        self.assertAlmostEqual(result[(50.0, 'a')]['b1'], 10.5)
+        self.assertAlmostEqual(result[(50.0, 'b')]['b1'], 21.0)
+        self.assertAlmostEqual(result[(50.0, 'c')]['b1'], 12.0)
+        self.assertAlmostEqual(result[(75.0, 'a')]['b1'], 11.25)
+        self.assertAlmostEqual(result[(75.0, 'b')]['b1'], 21.5)
+        self.assertAlmostEqual(result[(75.0, 'c')]['b1'], 17.5)
+        self.assertAlmostEqual(result[(100.0, 'a')]['b1'], 12.0)
+        self.assertAlmostEqual(result[(100.0, 'b')]['b1'], 22.0)
+        self.assertAlmostEqual(result[(100.0, 'c')]['b1'], 23.0)
+
+    def test_ancom_alt_percentiles(self):
+        table = pd.DataFrame([[12],
+                              [9],
+                              [1],
+                              [22],
+                              [20],
+                              [23]],
+                             index=['s1', 's2', 's3', 's4', 's5', 's6'],
+                             columns=['b1'])
+        grouping = pd.Series(['a', 'a', 'a', 'b', 'b', 'b'],
+                             index=['s1', 's2', 's3', 's4', 's5', 's6'])
+        result = ancom(table, grouping, percentiles=[42.0, 50.0])[1]
+        self.assertEqual(result.shape, (1, 4))
+        self.assertAlmostEqual(result[(42.0, 'a')]['b1'], 7.71999999)
+        self.assertAlmostEqual(result[(42.0, 'b')]['b1'], 21.68)
+        self.assertAlmostEqual(result[(50.0, 'a')]['b1'], 9.0)
+        self.assertAlmostEqual(result[(50.0, 'b')]['b1'], 22.0)
+
+        # order of percentiles in unimportant
+        result = ancom(table, grouping, percentiles=[50.0, 42.0])[1]
+        self.assertEqual(result.shape, (1, 4))
+        self.assertAlmostEqual(result[(42.0, 'a')]['b1'], 7.71999999)
+        self.assertAlmostEqual(result[(42.0, 'b')]['b1'], 21.68)
+        self.assertAlmostEqual(result[(50.0, 'a')]['b1'], 9.0)
+        self.assertAlmostEqual(result[(50.0, 'b')]['b1'], 22.0)
+
+    def test_ancom_no_percentiles(self):
+        table = pd.DataFrame([[12],
+                              [9],
+                              [1],
+                              [22],
+                              [20],
+                              [23]],
+                             index=['s1', 's2', 's3', 's4', 's5', 's6'],
+                             columns=['b1'])
+        grouping = pd.Series(['a', 'a', 'a', 'b', 'b', 'b'],
+                             index=['s1', 's2', 's3', 's4', 's5', 's6'])
+        result = ancom(table, grouping, percentiles=[])[1]
+        self.assertEqual(result.shape, (0, 0))
+
+    def test_ancom_invalid_percentiles(self):
```

Comment:
I know this is an annoying request, but can you please split this test case into several smaller test cases that each test one thing? Arrange, Act, Assert is the strategy we're using for new tests.